src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/design_patterns/DesignPattern.java

```diff
@@ -0,0 +1,22 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.design_patterns;
+
+
+public abstract class DesignPattern {
```

Comment:
In my opinion, refactor functionality and description & example should be divided into 2 separate classes - now DesignPattern has 2 responsibilites (code refactoring and documentation) which violates SRP